files/image_config/platform/rc.local

```diff
@@ -11,6 +11,22 @@
 #
 # By default this script does nothing.
 
+# If the machine.conf is absent, it indicates that the unit booted
+# into SONiC from another NOS. Extract the machine.conf from ONIE.
+if [ ! -e /host/machine.conf ]; then
+    onie_dev=$(blkid | grep ONIE-BOOT | head -n 1 | awk '{print $1}' |  sed -e 's/:.*$//')
+    mkdir -p /mnt/onie-boot
+    mount $onie_dev /mnt/onie-boot
+
+    if [ ! -e /mnt/onie-boot/onie/grub/grub-machine.cfg ]; then
+        echo "/mnt/onie-boot/onie/grub/grub-machine.cfg not found" >> /etc/migration.log
+    else
+        grep "=" /mnt/onie-boot/onie/grub/grub-machine.cfg > /host/machine.conf
```

Comment:
this is fragile.

check here. https://github.com/opencomputeproject/onie/pull/227/files

+if [ -r "$grub_machine_cfg" ] ; then
 +    . "$grub_machine_cfg"
 +    grep = $grub_machine_cfg | sed -e 's/onie_//' -e 's/=.*$//' | while read var ; do
 +        eval val='$'onie_$var
 +        printf "%-20s: %s\n" "ONIE ${var}" "$val"
 +    done
 +else